src/adapter/src/catalog/builtin.rs

```diff
@@ -1680,6 +1680,7 @@ pub static MZ_CLUSTER_REPLICA_STATUSES: Lazy<BuiltinTable> = Lazy::new(|| Builti
         .with_column("replica_id", ScalarType::String.nullable(false))
         .with_column("process_id", ScalarType::UInt64.nullable(false))
         .with_column("status", ScalarType::String.nullable(false))
+        .with_column("reason", ScalarType::String.nullable(true))
```

Comment:
Do these two need to be separate, or can they be folded in together? e.g. making them separate technically allows us to represent `status: "ready"` and `reason: "OOMKilled"`, which should be an invalid state?

Reply:
Indeed, but it also feels weird to have `ready`, `not-ready-oom-killed`, and `not-ready-other` as the possible values in a single column. The reason for non-readiness feels like a logically separate piece of information.

Reply:
I was thinking you could distill the reasons a bit, i.e. today you could have one of three statuses:
1. `ready`
2. `oom-killed`
3. `not-ready`

Where `oom-killed` implies "not ready". I totally see where you're coming from though w.r.t. non-readiness being a logically separate piece of info, so I could go either way